src/parsers/vmap.js

```diff
@@ -0,0 +1,114 @@
+import { Log } from '@clappr/core'
+import AdBreak from '../entities/ad-break'
+import { hmsToMilliseconds, xml2json } from '../utils'
+
+export default class VMAPManager {
+  /**
+   * Request VMAP XML.
+   * @param {String} URL ad server URL to request VMAP file.
+   * @returns {Promise} Promise resolved with VMAP parsed to plain text or one error.
+   */
+  request(url) {
+    return fetch(url)
+      .then(response => {
+        if (!response.ok) throw new Error('Network response was not ok')
+        return response.text()
+      })
+      .catch(error => Promise.reject(error))
+  }
+
+  /**
+   * Parses VMAP XML in plain text to JSON.
+   * @param {String} responseInPlainText Response of the fetch triggered for consuming VMAP.
+   * @returns {Promise} Promise resolved with raw AdBreaks list.
+   */
+  process(responseInPlainText) {
+    const parser = new DOMParser()
+    const xml = parser.parseFromString(responseInPlainText, 'application/xml')
+    const rawAdBreaks = xml2json(xml.documentElement)
+
+    return Promise.resolve(rawAdBreaks)
+  }
+
+  /**
+   * Receives one VMAP formatted response and return a list of AdBreaks instances.
+   * @param {Object} rawData VMAP formatted on JSON schema.
+   * @returns {Promises} An Array of all AdBreaks instances created or one error.
+   */
+  filterRawData(rawData) {
+    return rawData['@version'] && rawData['vmap:AdBreak']
+      ? this._filterStandardRawData(rawData)
+      : this._filterCustomRawData(rawData)
+  }
+
+  _filterStandardRawData(rawData) {
+    const adBreaks = []
+
+    try {
+      rawData['vmap:AdBreak'].forEach(item => {
```

Comment:
I tested with this VMAP [here](https://pubads.g.doubleclick.net/gampad/ads?sz=640x480&iu=/124319096/external/ad_rule_samples&ciu_szs=300x250&ad_rule=1&impl=s&gdfp_req=1&env=vp&output=vmap&unviewed_position_start=1&cust_params=deployment%3Ddevsite%26sample_ar%3Dpostonly&cmsid=496&vid=short_onecue&correlator=) with a single post-roll ad and the method raised an error since it is expecting an array of ad breaks but only a single one came.

Reply:
This [google sample page](https://developers.google.com/interactive-media-ads/docs/sdks/html5/client-side/tags) is a good reference for checking diffent combinations of VMAP. As far as I have tested only the single ad VMAPS are breaking. The rest is ok!

Reply:
I was just using these google examples to make this PR, but I didn't test all cases :facepalm:

Good catch, bug hunter! :gun: